File: src/main/java/by/auction/service/BetService.java
package by.auction.service;

import by.auction.entity.Bet;

import java.util.List;
import java.util.Optional;

/**
 * Service for data management in a table "bets"
 */
public interface BetService {

    /**
     * Find all bets
     * @return list of found bets
     */
    List<Bet> findAll();

    /**
     * Find bet by ID
     * @param id
     * @return Optional of found result
     */
    Optional<Bet> findById(Long id);

    /**
     * Find bets by auction ID
     * @param id
     * @return list of found bets
     */
    List<Bet> findByAuctionId(Long id);

    /**
     * Find bets by username
     * @param userName
     * @return list of found bets
     */
    List<Bet> findByUserName(String userName);

    /**
     * Save bet to DB
     * @param bet
     * @return saved bet
     */
    Bet save(final Bet bet);

    /**
     * Delete bet by ID
     * @param id
     */
    void deleteById(Long id);

}

File: auction-client-angular/src/app/services/bet/bet.service.ts
import {Injectable} from '@angular/core';
import {HttpClient, HttpParams} from '@angular/common/http';
import {Observable} from 'rxjs/Observable';
import {Bet} from '../../models/bet';
import {ConfigService} from "../config/config.service";

/**
 * Service to access bets data
 */
@Injectable()
export class BetService {

  /**
   * Constructor for bet service
   * @param {HttpClient} http
   * @param {ConfigService} config
   */
  constructor(private http: HttpClient,
              private config: ConfigService) {
  }

  /**
   * Get api uri
   * @returns {string}
   */
  uri() {
    return this.config.getApiHref() + 'bets';
  }

  /**
   * Find all bets
   * @returns {Observable<Bet[]>}
   */
  findAll(): Observable<Bet[]> {
    return this.http.get<Bet[]>(this.uri());
  }

  /**
   * Find bet by id
   * @param id
   * @returns {Observable<Bet>}
   */
  findById(id): Observable<Bet> {
    return this.http.get<Bet>(this.uri(), {
      params: new HttpParams().set('id', id)
    });
  }

  /**
   * Find bets by auction id
   * @param id
   * @returns {Observable<Bet[]>}
   */
  findByAuctionId(id): Observable<Bet[]> {
    return this.http.get<Bet[]>(this.uri(), {
      params: new HttpParams().set('auctionId', id)
    });
  }

  /**
   * Find bets by username
   * @param user
   * @returns {Observable<Bet[]>}
   */
  findByUsername(user): Observable<Bet[]> {
    return this.http.get<Bet[]>(this.uri(), {
      params: new HttpParams().set('username', user)
    });
  }

  /**
   * Save bet
   * @param {Bet} bet
   * @returns {Observable<Bet>}
   */
  save(bet: Bet): Observable<Bet> {
    return this.http.post<Bet>(this.uri(), bet);
  }

  /**
   * Delete bet by id
   * @param id
   * @returns {Observable<Object>}
   */
  deleteById(id) {
    return this.http.delete(this.uri(), {
      params: new HttpParams().set('delete', id)
    });
  }

}
